src/bin/miri.rs

```diff
@@ -78,14 +78,22 @@ impl rustc_driver::Callbacks for MiriCompilerCalls {
 
             let early_dcx = EarlyDiagCtxt::new(tcx.sess.opts.error_format);
             init_late_loggers(&early_dcx, tcx);
-            if !tcx.crate_types().contains(&CrateType::Executable) {
```

Comment:
I think this restriction should stay, it doesn't make a lot of sense to execute a library.